***
# <font color=green size=10>CURSO DE ESTATÍSTICA - PARTE 2</font>
***

# <font color=green>1 CONHECENDO OS DADOS</font>
***

## <font color=green>1.1 Dataset do projeto</font>
***

### Pesquisa Nacional por Amostra de Domicílios - 2015

A **Pesquisa Nacional por Amostra de Domicílios - PNAD** investiga anualmente, de forma permanente, características gerais da população, de educação, trabalho, rendimento e habitação e outras, com periodicidade variável, de acordo com as necessidades de informação para o país, como as características sobre migração, fecundidade, nupcialidade, saúde, segurança alimentar, entre outros temas. O levantamento dessas estatísticas constitui, ao longo dos 49 anos de realização da pesquisa, um importante instrumento para formulação, validação e avaliação de políticas orientadas para o desenvolvimento socioeconômico e a melhoria das condições de vida no Brasil.

### Fonte dos Dados

https://ww2.ibge.gov.br/home/estatistica/populacao/trabalhoerendimento/pnad2015/microdados.shtm

### Variáveis utilizadas

> ### Renda
> ***

Rendimento mensal do trabalho principal para pessoas de 10 anos ou mais de idade.

> ### Idade
> ***

Idade do morador na data de referência em anos.

> ### Altura (elaboração própria)
> ***

Altura do morador em metros.

> ### UF
> ***

|Código|Descrição|
|---|---|
|11|Rondônia|
|12|Acre|
|13|Amazonas|
|14|Roraima|
|15|Pará|
|16|Amapá|
|17|Tocantins|
|21|Maranhão|
|22|Piauí|
|23|Ceará|
|24|Rio Grande do Norte|
|25|Paraíba|
|26|Pernambuco|
|27|Alagoas|
|28|Sergipe|
|29|Bahia|
|31|Minas Gerais|
|32|Espírito Santo|
|33|Rio de Janeiro|
|35|São Paulo|
|41|Paraná|
|42|Santa Catarina|
|43|Rio Grande do Sul|
|50|Mato Grosso do Sul|
|51|Mato Grosso|
|52|Goiás|
|53|Distrito Federal|

> ### Sexo	
> ***

|Código|Descrição|
|---|---|
|0|Masculino|
|1|Feminino|

> ### Anos de Estudo
> ***

|Código|Descrição|
|---|---|
|1|Sem instrução e menos de 1 ano|
|2|1 ano|
|3|2 anos|
|4|3 anos|
|5|4 anos|
|6|5 anos|
|7|6 anos|
|8|7 anos|
|9|8 anos|
|10|9 anos|
|11|10 anos|
|12|11 anos|
|13|12 anos|
|14|13 anos|
|15|14 anos|
|16|15 anos ou mais|
|17|Não determinados| 
||Não aplicável|

> ### Cor
> ***

|Código|Descrição|
|---|---|
|0|Indígena|
|2|Branca|
|4|Preta|
|6|Amarela|
|8|Parda|
|9|Sem declaração|

#### <font color='red'>Observação</font>
***
> Os seguintes tratamentos foram realizados nos dados originais:
> 1. Foram eliminados os registros onde a **Renda** era inválida (999 999 999 999);
> 2. Foram eliminados os registros onde a **Renda** era missing;
> 3. Foram considerados somente os registros das **Pessoas de Referência** de cada domicílio (responsável pelo domicílio).

### Importando pandas e lendo o dataset do projeto

https://pandas.pydata.org/

In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('dados.csv')

---
---

# <font color=green>2 DISTRIBUIÇÕES DE PROBABILIDADE</font>
***

## <font color=red>Problema</font>
***

Em um concurso para preencher uma vaga de cientista de dados temos um total de **10 questões** de múltipla escolha com **3 alternativas possíveis** em cada questão. **Cada questão tem o mesmo valor.** Suponha que um candidato resolva se aventurar sem ter estudado absolutamente nada. Ele resolve fazer a prova de olhos vendados e chutar todas as resposta. Assumindo que a prova **vale 10 pontos e a nota de corte seja 5**, obtenha a probabilidade deste candidato **acertar 5 questões** e também a probabilidade deste candidato **passar para a próxima etapa do processo seletivo**.

## <font color=green>2.1 Distribuição Binomial</font>
***

Um evento **binomial** é caracterizado pela possibilidade de ocorrência de apenas duas categorias. Estas categorias somadas representam todo o espaço amostral, sendo também mutuamente excludentes, ou seja, a ocorrência de uma implica na não ocorrência da outra.

Em análises estatísticas o uso mais comum da distribuição binomial é na solução de problemas que envolvem situações de **sucesso** e **fracasso**.

# $$P(k)=\binom{n}{k} p^k q^{n-k}$$

Onde:

$p$ = probabilidade de sucesso

$q = (1 - p)$ = probabilidade de fracasso

$n$ = número de eventos estudados

$k$ = número de eventos desejados que tenham sucesso

### Experimento Binomial

1. Realização de $n$ ensaios idênticos.

2. Os ensaios são independentes.

3. Somente dois resultados são possíveis, exemplo: Verdadeiro ou falso; Cara ou coroa; Sucesso ou fracasso.

4. A probabilidade de sucesso é representada por $p$ e a de fracasso por $1-p=q$. Estas probabilidades não se modificam de ensaio para ensaio.

### Média da distribuição binomial

O valor esperado ou a média da distribuição binomial é igual ao número de experimentos realizados multiplicado pela chance de ocorrência do evento.

# $$\mu = n \times p$$

### Desvio padrão da distribuição binomial

O desvio padrão é o produto entre o número de experimentos, a probabilidade de sucesso e a probabilidade de fracasso.

# $$\sigma = \sqrt{n \times p \times q}$$

### Importando bibliotecas
https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.comb.html

In [3]:
from scipy.special import comb

### Combinações

Número de combinações de $n$ objetos, tomados $k$ a cada vez, é:

# $$C_{k}^{n} = \binom{n}{k} = \frac{n!}{k!(n - k)!}$$

Onde

## $$n! = n\times(n-1)\times(n-2)\times...\times(2)\times(1)$$
## $$k! = k\times(k-1)\times(k-2)\times...\times(2)\times(1)$$

Por definição

## $$0! = 1$$

## <font color='blue'>Exemplo: Mega Sena</font>

Em um volante de loteria da Mega Sena temos um total de **60 números** para escolher onde a aposta mínima é de **seis números**. Você que é curiosa(o) resolve calcular a probabilidade de se acertar na Mega Sena com apenas **um jogo**. Para isso precisamos saber quantas **combinações de seis números podem ser formadas com os 60 números disponíveis**.

### $$C_{6}^{60} = \binom{60}{6} = \frac{60!}{6!(60 - 6)!}$$

In [4]:
combinacoes = comb(60,6)
combinacoes

50063860.0

In [5]:
# probabilidade de acertar 1 bilhete = 1 jogo de 6 numeros
probabilidade = 1 / combinacoes
print('%0.15f' % probabilidade)


0.000000019974489


## <font color='blue'>Exemplo: Concurso para cientista de dados</font>

Em um concurso para preencher uma vaga de cientista de dados temos um total de **10 questões** de múltipla escolha com **3 alternativas possíveis** em cada questão. **Cada questão tem o mesmo valor.** Suponha que um candidato resolva se aventurar sem ter estudado absolutamente nada. Ele resolve fazer a prova de olhos vendados e chutar todas as resposta. Assumindo que a prova **vale 10 pontos e a nota de corte seja 5**, obtenha a probabilidade deste candidato **acertar 5 questões** e também a probabilidade deste candidato **passar para a próxima etapa do processo seletivo**.

### Qual o número de ensaios ($n$)?

In [12]:
n = 10 # numero de questoes

### Os ensaios são independentes?

Sim. A opção escolhida em uma questão não influencia em nada a opção escolhida em outra questão.

### Somente dois resultados são possíveis em cada ensaio?

Sim. O candidato tem duas possibilidades, ACERTA ou ERRAR uma questão. 

### Qual a probabilidade de sucesso ($p$)?

In [8]:
numero_alternativa_questao = 3
p = 1 / numero_alternativa_questao # acertar 1 em 3
p 

0.3333333333333333

### Qual a probabilidade de fracasso ($q$)?

In [9]:
q = 1 - p
q

0.6666666666666667

### Qual o total de eventos que se deseja obter sucesso ($k$)?

In [6]:
# 5 = precisa acertar pelo menos 5
k = 5 


### Solução 1

In [13]:
probabilidade = (comb(n,k) * (p ** k) * (q**(n-k)) )
print('%0.8f' % probabilidade)

0.13656455


In [11]:
# em % 
print('%0.2f' % (probabilidade * 100) )

13.66


### Importando bibliotecas
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.binom.html

In [4]:
# fazendo de forma mais simples
from scipy.stats import binom

### Solução 2

In [14]:
probabilidade = binom.pmf(k,n,p)
print('%0.8f' % probabilidade)

0.13656455


### Obter a probabilidade do candidato passar

### $$P(acertar \geq 5) = P(5) + P(6) + P(7) + P(8) + P(9) + P10)$$

In [14]:
probabilidade = binom.pmf(5,n,p)+binom.pmf(6,n,p)+binom.pmf(7,n,p)+binom.pmf(8,n,p)+binom.pmf(9,n,p)+binom.pmf(10,n,p)
print('%0.8f' % probabilidade)

0.21312808


In [15]:
# passando uma lista
binom.pmf([5,6,7,8,9,10],n,p).sum()

0.21312808006909492

In [17]:
# usando uma funcao cumulativa ou seja = Probab de acertar 1 + probab acertar 2 + ... + prob acertar 4
probabilidade_acertar_4 = binom.cdf(4,n,p)
probabilidade_acertar_4
# entao a probabilidade do individuo passar = 1 - probabilidade_acertar_4
1 - probabilidade_acertar_4

0.21312808006909523

In [18]:
# a funcao sf faz o calculo acima direto
binom.sf(4,n,p)

0.21312808006909525

As perguntas a respeito dos cálculos acima são descritos normalmente assim:

- calcule a probabilidade de ter pelo menos 5 acertos > P(5)+P(6)....+ P(10)
- calcule a probabilidade de ter no máximo 4 acertos  > P(0)+P(1)....+ P(4)

Exercicios

Uma moeda, perfeitamente equilibrada, é lançada para o alto quatro vezes. Utilizando a distribuição binomial, obtenha a probabilidade de a moeda cair com a face coroa voltada para cima duas vezes.

In [29]:
n = 4 # numero ensaios
p = 1/2 # prob sucesso - sair coroa - (ver de forma unitaria) - acertar 1 em 2
q = 1 - p # prob fracasso
k = 2 # total de eventos que se deseja obter sucesso > dado pelo problema: jogando uma moeda 4x - qual a probabilidade de dar coroa 2x
probabilidade = binom.pmf(k,n,p)
print('%0.8f' % probabilidade)


0.37500000


Um dado, perfeitamente equilibrado, é lançado para o alto dez vezes. Utilizando a distribuição binomial, obtenha a probabilidade de o dado cair com o número cinco voltado para cima pelo menos três vezes.

In [28]:
n = 10  # numero de eventos
p = 1 / 6 # prob de obter o valor desejado
q = 1 -p  # prob de não obter o valor desejado
k = 3 # numero de eventos sucesso
probabilidade = binom.pmf(k,n,p)
print('%0.8f' % probabilidade + " errado !!") #15,5%  
# calculo acima esta errado - calcula a probabilidade de obter o numero 5 três vezes - o enunciado pede pelo menos 3x
x = binom.sf(2,n,p)
print('%0.8f' % x)
# ou 
y = binom.pmf([3,4,5,6,7,8,9,10],n,p).sum()
print('%0.8f' % y)

0.15504536 errado !!
0.22477320
0.22477320


## <font color='blue'>Exemplo: Gincana</font>

Uma cidade do interior realiza todos os anos uma gincana para arrecadar fundos para o hospital da cidade. Na última gincana se sabe que a **proporção de participantes do sexo feminino foi de 60%**. **O total de equipes, com 12 integrantes, inscritas na gincana deste ano é de 30**. Com as informações acima responda: Quantas equipes deverão ser formadas por **8 mulheres**?

Pense antes de continuar lendo .....

qual a probabilidade de termos uma equipe 8 mulheres e 4 homens ?

experimento binominal = seleção de homens e mulheres para fazerem parte de uma equipe !! - sabemos que temos 60% mulheres e 40% homens - temos que selecionar uma pessoa 12 vezes - a probabilidade de sucesso = 60% porque estamos considerando mulheres. Temos:

n = 12
p = 0,6
q = 0,4
k = 8

### Solução

In [34]:
p = 0.6
n = 12
k = 8
probabilidade = binom.pmf(k,n,p)
probabilidade

0.21284093952

Temos 21.28% de chance de uma equipe ser selecionada com uma equipe de 8 mulheres e 4 homens !! Mas o que o problema pergunta é quantas das equipes serão formadas por esse tipo de equipe ? temos que usar o conceito de média !! u = n * p (fórmula acima >> n = 30 equipes)

In [35]:
equipes = 30 * probabilidade
equipes

6.3852281856

Suponha que a probabilidade de um casal ter filhos com olhos azuis seja de 22%. Em 50 famílias, com 3 crianças cada uma, quantas podemos esperar que tenham dois filhos com olhos azuis?

In [36]:
p = 0.22
q = 1 - p
k = 2
n = 3
prob = binom.pmf(k,n,p)
prob

0.11325599999999993

a probabilidade de uma familia ter 2 filhos de olhos azuis é de 11.32%, mas qual a média em 50 familias ?

In [37]:
media = prob * 50
media

5.662799999999996

---
---

## <font color=red>Problema</font>
***

Um restaurante recebe em média **20 pedidos por hora**. Qual a chance de que, em determinada hora escolhida ao acaso, o restaurante receba **15 pedidos**?

## <font color=green>2.2 Distribuição Poisson</font>
***

**É empregada para descrever o número de ocorrências em um intervalo de tempo ou espaço específico. Os eventos são caracterizados pela possibilidade de contagem dos sucessos, mas a não possibilidade de contagem dos fracassos.**

Como exemplos de processos onde podemos aplicar a distribuição de Poisson temos a determinação do número de clientes que entram em uma loja em determinada hora, o número de carros que chegam em um drive-thru de uma lanchonete na hora do almoço, a determinação do número de acidentes registrados em um trecho de estrada etc.

# $$P(k) = \frac{e^{-\mu}(\mu)^k}{k!}$$

Onde:

$e$ = constante cujo valor aproximado é 2,718281828459045

$\mu$ = representa o número médio de ocorrências em um determinado intervalo de tempo ou espaço

$k$ = número de sucessos no intervalo desejado

### Experimento Poisson

1. A probabilidade de uma ocorrência é a mesma em todo o intervalo observado.

2. O número de ocorrências em determinado intervalo é independente do número de ocorrências em outros intervalos.

3. A probabilidade de uma ocorrência é a mesma em intervalos de igual comprimento.

### Média da distribuição Poisson

# $$\mu$$

### Desvio padrão da distribuição Poisson

# $$\sigma = \sqrt{\mu}$$

### Importando bibliotecas

http://www.numpy.org/

In [27]:
import numpy as np

In [16]:
np.e

2.718281828459045

## <font color='blue'>Exemplo: Delivery</font>

Um restaurante recebe em média **20 pedidos por hora**. Qual a chance de que, em determinada hora escolhida ao acaso, o restaurante receba **15 pedidos**?

### Qual o número médio de ocorrências por hora ($\mu$)?

In [20]:
u = 20

### Qual o número de ocorrências que queremos obter no período ($k$)?

In [19]:
k = 15

### Solução 1

In [21]:
# solucao manual - formula
probabilidade = ((np.e ** (-u)) * (u ** k)) / (np.math.factorial(k))
probabilidade
print('%0.8f' % probabilidade) # 5% de chance de receber 15 pedidos

0.05164885


### Importando bibliotecas

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.poisson.html

### Solução 2

In [22]:
from scipy.stats import poisson

probabilidade = poisson.pmf(k, u)
probabilidade # mesmo valor = 5% de chance

0.05164885353175814

O número médio de clientes que entram em uma padaria por hora é igual a 20. Obtenha a probabilidade de, na próxima hora, entrarem exatamente 25 clientes.

In [45]:
u = 20 # media
k = 25 
poisson.pmf(k,u) # probabilidade 4,46%

0.04458764910826764

---
---

## <font color=red>Problema</font>
***

Em um estudo sobre as alturas dos moradores de uma cidade verificou-se que o conjunto de dados segue uma **distribuição aproximadamente normal**, com **média 1,70** e **desvio padrão de 0,1**. Com estas informações obtenha o seguinte conjunto de probabilidades:

> **A.** probabilidade de uma pessoa, selecionada ao acaso, ter menos de 1,80 metros.

> **B.** probabilidade de uma pessoa, selecionada ao acaso, ter entre 1,60 metros e 1,80 metros.    

> **C.** probabilidade de uma pessoa, selecionada ao acaso, ter mais de 1,90 metros.

## <font color=green>2.3 Distribuição Normal</font>
***

A distribuição normal é uma das mais utilizadas em estatística. É uma distribuição contínua, onde a distribuição de frequências de uma variável quantitativa apresenta a forma de sino e é simétrica em relação a sua média.

![Normal](https://caelum-online-public.s3.amazonaws.com/1178-estatistica-parte2/01/img001.png)

### Características importantes

1. É simétrica em torno da média;

2. A área sob a curva corresponde à proporção 1 ou 100%;

3. As medidas de tendência central (média, mediana e moda) apresentam o mesmo valor;

4. Os extremos da curva tendem ao infinito em ambas as direções e, teoricamente, jamais tocam o eixo $x$;

5. O desvio padrão define o achatamento e largura da distribuição. Curvas mais largas e mais achatadas apresentam valores maiores de desvio padrão;

6. A distribuição é definida por sua média e desvio padrão;

7. A probabilidade sempre será igual à área sob a curva, delimitada pelos limites inferior e superior.

# $$f(x) = \frac{1}{\sqrt{2\pi\sigma}}e^{-\frac{1}{2}\left(\frac{x-\mu}{\sigma}\right)^2}$$

Onde:

$x$ = variável normal

$\sigma$ = desvio padrão

$\mu$ = média

A probabilidade é obtida a partir da área sob a curva, delimitada pelos limites inferior e superior especificados. Um exemplo pode ser visto na figura abaixo.

![alt text](https://caelum-online-public.s3.amazonaws.com/1178-estatistica-parte2/01/img002.png)


Para obter a área acima basta calcular a integral da função para os intervalos determinados. Conforme equação abaixo:

# $$P(L_i<x<L_s) = \int_{L_i}^{L_s}\frac{1}{\sqrt{2\pi\sigma}}e^{-\frac{1}{2}\left(\frac{x-\mu}{\sigma}\right)^2}$$

Onde:

$x$ = variável normal

$\sigma$ = desvio padrão

$\mu$ = média

$L_i$ = limite inferior

$L_s$ = limite superior

### Tabelas padronizadas

As tabelas padronizadas foram criadas para facilitar a obtenção dos valores das áreas sob a curva normal e eliminar a necessidade de solucionar integrais definidas.

Para consultarmos os valores em uma tabela padronizada basta transformarmos nossa variável em uma variável padronizada $Z$.

Esta variável $Z$ representa o afastamento em desvios padrões de um valor da variável original em relação à média.

# $$Z = \frac{x-\mu}{\sigma}$$

Onde:

$x$ = variável normal com média $\mu$ e desvio padrão $\sigma$

$\sigma$ = desvio padrão

$\mu$ = média

### Construindo tabela normal padronizada
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.norm.html

In [34]:
import pandas as pd
import numpy as np
from scipy.stats import norm

tabela_normal_padronizada = pd.DataFrame(
    [], 
    index=["{0:0.2f}".format(i / 100) for i in range(0, 400, 10)],
    columns = ["{0:0.2f}".format(i / 100) for i in range(0, 10)])

for index in tabela_normal_padronizada.index:
    for column in tabela_normal_padronizada.columns:
        Z = np.round(float(index) + float(column), 2)
        tabela_normal_padronizada.loc[index, column] = "{0:0.4f}".format(norm.cdf(Z))

tabela_normal_padronizada.rename_axis('Z', axis = 'columns', inplace = True)

tabela_normal_padronizada

Z       0.00    0.01    0.02    0.03    0.04    0.05    0.06    0.07    0.08  \
0.00  0.5000  0.5040  0.5080  0.5120  0.5160  0.5199  0.5239  0.5279  0.5319   
0.10  0.5398  0.5438  0.5478  0.5517  0.5557  0.5596  0.5636  0.5675  0.5714   
0.20  0.5793  0.5832  0.5871  0.5910  0.5948  0.5987  0.6026  0.6064  0.6103   
0.30  0.6179  0.6217  0.6255  0.6293  0.6331  0.6368  0.6406  0.6443  0.6480   
0.40  0.6554  0.6591  0.6628  0.6664  0.6700  0.6736  0.6772  0.6808  0.6844   
0.50  0.6915  0.6950  0.6985  0.7019  0.7054  0.7088  0.7123  0.7157  0.7190   
0.60  0.7257  0.7291  0.7324  0.7357  0.7389  0.7422  0.7454  0.7486  0.7517   
0.70  0.7580  0.7611  0.7642  0.7673  0.7704  0.7734  0.7764  0.7794  0.7823   
0.80  0.7881  0.7910  0.7939  0.7967  0.7995  0.8023  0.8051  0.8078  0.8106   
0.90  0.8159  0.8186  0.8212  0.8238  0.8264  0.8289  0.8315  0.8340  0.8365   
1.00  0.8413  0.8438  0.8461  0.8485  0.8508  0.8531  0.8554  0.8577  0.8599   
1.10  0.8643  0.8665  0.8686  0.8708  0.8729  0.8749  0.8770  0.8790  0.8810   
1.20  0.8849  0.8869  0.8888  0.8907  0.8925  0.8944  0.8962  0.8980  0.8997   
1.30  0.9032  0.9049  0.9066  0.9082  0.9099  0.9115  0.9131  0.9147  0.9162   
1.40  0.9192  0.9207  0.9222  0.9236  0.9251  0.9265  0.9279  0.9292  0.9306   
1.50  0.9332  0.9345  0.9357  0.9370  0.9382  0.9394  0.9406  0.9418  0.9429   
1.60  0.9452  0.9463  0.9474  0.9484  0.9495  0.9505  0.9515  0.9525  0.9535   
1.70  0.9554  0.9564  0.9573  0.9582  0.9591  0.9599  0.9608  0.9616  0.9625   
1.80  0.9641  0.9649  0.9656  0.9664  0.9671  0.9678  0.9686  0.9693  0.9699   
1.90  0.9713  0.9719  0.9726  0.9732  0.9738  0.9744  0.9750  0.9756  0.9761   
2.00  0.9772  0.9778  0.9783  0.9788  0.9793  0.9798  0.9803  0.9808  0.9812   
2.10  0.9821  0.9826  0.9830  0.9834  0.9838  0.9842  0.9846  0.9850  0.9854   
2.20  0.9861  0.9864  0.9868  0.9871  0.9875  0.9878  0.9881  0.9884  0.9887   
2.30  0.9893  0.9896  0.9898  0.9901  0.9904  0.9906  0.9909  0.9911  0.9913   
2.40  0.9918  0.9920  0.9922  0.9925  0.9927  0.9929  0.9931  0.9932  0.9934   
2.50  0.9938  0.9940  0.9941  0.9943  0.9945  0.9946  0.9948  0.9949  0.9951   
2.60  0.9953  0.9955  0.9956  0.9957  0.9959  0.9960  0.9961  0.9962  0.9963   
2.70  0.9965  0.9966  0.9967  0.9968  0.9969  0.9970  0.9971  0.9972  0.9973   
2.80  0.9974  0.9975  0.9976  0.9977  0.9977  0.9978  0.9979  0.9979  0.9980   
2.90  0.9981  0.9982  0.9982  0.9983  0.9984  0.9984  0.9985  0.9985  0.9986   
3.00  0.9987  0.9987  0.9987  0.9988  0.9988  0.9989  0.9989  0.9989  0.9990   
3.10  0.9990  0.9991  0.9991  0.9991  0.9992  0.9992  0.9992  0.9992  0.9993   
3.20  0.9993  0.9993  0.9994  0.9994  0.9994  0.9994  0.9994  0.9995  0.9995   
3.30  0.9995  0.9995  0.9995  0.9996  0.9996  0.9996  0.9996  0.9996  0.9996   
3.40  0.9997  0.9997  0.9997  0.9997  0.9997  0.9997  0.9997  0.9997  0.9997   
3.50  0.9998  0.9998  0.9998  0.9998  0.9998  0.9998  0.9998  0.9998  0.9998   
3.60  0.9998  0.9998  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999   
3.70  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999   
3.80  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999   
3.90  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000   

Z       0.09  
0.00  0.5359  
0.10  0.5753  
0.20  0.6141  
0.30  0.6517  
0.40  0.6879  
0.50  0.7224  
0.60  0.7549  
0.70  0.7852  
0.80  0.8133  
0.90  0.8389  
1.00  0.8621  
1.10  0.8830  
1.20  0.9015  
1.30  0.9177  
1.40  0.9319  
1.50  0.9441  
1.60  0.9545  
1.70  0.9633  
1.80  0.9706  
1.90  0.9767  
2.00  0.9817  
2.10  0.9857  
2.20  0.9890  
2.30  0.9916  
2.40  0.9936  
2.50  0.9952  
2.60  0.9964  
2.70  0.9974  
2.80  0.9981  
2.90  0.9986  
3.00  0.9990  
3.10  0.9993  
3.20  0.9995  
3.30  0.9997  
3.40  0.9998  
3.50  0.9998  
3.60  0.9999  
3.70  0.9999  
3.80  0.9999  
3.90  1.0000

<img src='https://caelum-online-public.s3.amazonaws.com/1178-estatistica-parte2/01/img003.png' width='250px'>

A tabela acima fornece a área sob a curva entre $-\infty$ e $Z$ desvios padrão acima da média. Lembrando que por se tratar de valores padronizados temos $\mu = 0$.

## <font color='blue'>Exemplo: Qual sua altura?</font>

Em um estudo sobre as alturas dos moradores de uma cidade verificou-se que o conjunto de dados segue uma **distribuição aproximadamente normal**, com **média 1,70** e **desvio padrão de 0,1**. Com estas informações obtenha o seguinte conjunto de probabilidades:

> **A.** probabilidade de uma pessoa, selecionada ao acaso, ter menos de 1,80 metros.

> **B.** probabilidade de uma pessoa, selecionada ao acaso, ter entre 1,60 metros e 1,80 metros.    

> **C.** probabilidade de uma pessoa, selecionada ao acaso, ter mais de 1,90 metros.

### Problema A - Identificação da área sob a curva

<img style='float: left' src='https://caelum-online-public.s3.amazonaws.com/1178-estatistica-parte2/01/img004.png' width='350px'>

### Obter a variável padronizada $Z$

In [23]:
# lembrando z = (x - u) / desvio_padrao
x = 1.8
u = 1.7
d = 0.1
z = (x - u) / d
z


1.0000000000000009

### Solução 1 - Utilizando tabela

Acima Z padronizado = 1 consultando a tabela acima Z = 1 correponde a 0,8413 !! 
Logo a probabilidade de uma pessoa, selecionada ao acaso, ter menos de 1,80 metros é 84,13%  !!


### Solução 2 - Utilizando Scipy

In [24]:
from scipy.stats import norm
norm.cdf(z) ## basta informar o Z padronizado - não precisamos ficar olhando tabela

0.8413447460685431

### Problema B - Identificação da área sob a curva

<img style='float: left' src='https://caelum-online-public.s3.amazonaws.com/1178-estatistica-parte2/01/img005.png' width='350px'>

> **B.** probabilidade de uma pessoa, selecionada ao acaso, ter entre 1,60 metros e 1,80 metros.    


### Obter a variável padronizada $Z$

### Solução 1 - Utilizando tabela

Entender que:
a - de menos infinito a 1,70 = temos de 0,5 de probabilidade, conferir > z = 1,7 - 1,7 / 0,1 => z = 0 (ver tabela)
b - que a área de 1,60 a 1,70 é igual a aŕea de 1,70 a 1,80

então podemos dizer que a área de 1,7 a 1,8 é : a area de 1,8 - 0,5 (area de menos infinito a 1,70)

In [25]:
z_180 = (1.8 - 1.7) / 0.1
print("Z normalizado para 1,80 = ", z_180)
area_180 = norm.cdf(z_180)
print("Area de menos infinito a 1.80 = ", area_180)
area_170_180 = area_180 - 0.5
print("Area de 1,8 (0,8413) a 1,7 (0,5) = ", area_170_180)
area_160_180 = area_170_180 * 2
print("Area de 1,60 a 1,80 (0,3413 * 2)= ", area_160_180)

Z normalizado para 1,80 =  1.0000000000000009
Area de menos infinito a 1.80 =  0.8413447460685431
Area de 1,8 (0,8413) a 1,7 (0,5) =  0.34134474606854315
Area de 1,60 a 1,80 (0,3413 * 2)=  0.6826894921370863


68,26% probabilidade de uma pessoa, selecionada ao acaso, ter entre 1,60 metros e 1,80 metros.    

### Solução 2 - Utilizando Scipy

In [26]:
Z_superior = (1.8 - 1.7) / 0.1
probabilidade_sup = norm.cdf(Z_superior)
probabilidade_sup

0.8413447460685431

In [27]:
lado_dir = 1 - probabilidade_sup
prob_160_180 = probabilidade_sup - lado_dir
prob_160_180 

0.6826894921370863

Mesmo resultado das celulas anteriores.

### Problema C - Identificação da área sob a curva

<img style='float: left' src='https://caelum-online-public.s3.amazonaws.com/1178-estatistica-parte2/01/img006.png' width='350px'>

> **C.** probabilidade de uma pessoa, selecionada ao acaso, ter mais de 1,90 metros.

### Obter a variável padronizada $Z$

In [31]:
Z = (1.9 - 1.7) / 0.1
prob = norm.cdf(Z)
prob 


0.9772498680518208

### Solução 1 - Utilizando tabela

In [32]:
area_apos_190 = 1 - prob 
area_apos_190

0.02275013194817921

### Solução 2 - Utilizando Scipy

In [33]:
prob = 1 - norm.cdf(Z)
prob

0.02275013194817921

In [36]:
## O cdf É ESPERTO SE VC PASSAR O Z NEGATIVO - ELE JA FAZ A CONTA ACIMA DIRETO - 
## ELE ENTENDE QUE É PARA CALCULAR A AREA RESTANTE DE z
prob = norm.cdf(-Z)
prob

0.022750131948179216

EXERCICIO

O faturamento diário de um motorista de aplicativo segue uma distribuição aproximadamente normal, com média R$ 300,00 e desvio padrão igual a R$ 50,00. Obtenha as probabilidades de que, em um dia aleatório, o motorista ganhe:

1) Entre R$ 250,00 e R$ 350,00

2) Entre R$ 400,00 e R$ 500,00

In [74]:
# entre 250 e 350
z = 1
prob_sup = norm.cdf(z)
print("prob superior = ", prob_sup)
prob_intervalo_dir = 1 - prob_sup
print("prob intervalo dir = ", prob_intervalo_dir)
prob_intervalo_pedido = prob_sup - prob_intervalo_dir
prob_intervalo_pedido


prob superior =  0.8413447460685429
prob intervalo dir =  0.15865525393145707


0.6826894921370859

In [77]:
# entre 400 e 500
z_sup = (500-300) / 50
z_inf = (400-300) / 50
prob = norm.cdf(z_sup) - norm.cdf(z_inf)
prob

0.02271846070634609

In [78]:
# entre 400 e 500 - 2 forma
z = (500-300) / 50
prob_sup = norm.cdf(z)
z_inf = (400-300) / 50
prob_inf = norm.cdf(z_inf)
print("prob superior = ", prob_sup)
print("prob inferior = ", prob_inf)
prob_intervalo = prob_sup - prob_inf
prob_intervalo 

prob superior =  0.9999683287581669
prob inferior =  0.9772498680518208


0.02271846070634609

A aplicação de uma prova de estatística em um concurso apresentou um conjunto de notas normalmente distribuídas. Verificou-se que o conjunto de notas tinha média 70 e desvio padrão de 5 pontos.

Qual a probabilidade de um aluno, selecionado ao acaso, ter nota menor que 85?


In [ ]:
z = (85 - 70) / 5
norm.cdf(z)

0.9986501019683699

O Inmetro verificou que as lâmpadas incandescentes da fabricante XPTO apresentam uma vida útil normalmente distribuída, com média igual a 720 dias e desvio padrão igual a 30 dias. Calcule a probabilidade de uma lâmpada, escolhida ao acaso, durar:

1) Entre 650 e 750 dias

2) Mais que 800 dias

3) Menos que 700 dias

In [38]:
media = 720
desvio = 30

# 1
z_inf = (650 - media) / desvio 
z_sup = (750 - media) / desvio 
p_inf = norm.cdf(z_inf)
p_sup = norm.cdf(z_sup)
p_intervalo = p_sup - p_inf 
print("Probabilidade entre 650 - 750 = ", p_intervalo)
# 2
z2 = (800 - media) / desvio
p2 = norm.cdf(-z2)
print("probabilidade maior que 800 dias = ", p2)
# 3
z3 = (700 - media) / desvio
p3 = norm.cdf(z3)
print("probabilidade menor que 700 dias = ", p3)

Probabilidade entre 650 - 750 =  0.8315294174398976
probabilidade maior que 800 dias =  0.0038303805675897365
probabilidade menor que 700 dias =  0.2524925375469229


Utilizando a tabela padronizada, ou o ferramental disponibilizado pelo Python, encontre a área sob a curva normal para os valores de Z abaixo:

1) Z < 1,96

2) Z > 2,15

3) Z < -0,78

4) Z > 0,59

In [44]:
# 1
prob1 = norm.cdf(1.96)
print(prob1)
# 2
prob2 = norm.cdf(-2.15)
print(prob2)
# 3
prob3 = norm.cdf(-0.78)
print(prob3)
# 4
prob4 = norm.cdf(-0.59)
print(prob4)

0.9750021048517795
0.0157776073910905
0.21769543758573312
0.27759532475346493


# <font color=green>3 AMOSTRAGEM</font>
***

## <font color=green>3.1 População e Amostra</font>
***

### População
Conjunto de todos os elementos de interesse em um estudo. Diversos elementos podem compor uma população, por exemplo: pessoas, idades, alturas, carros etc.

Com relação ao tamanho, as populações podem ser limitadas (populações finitas) ou ilimitadas (populações infinitas).

### Populações finitas

Permitem a contagem de seus elementos. Como exemplos temos o número de funcionário de uma empresa, a quantidade de alunos em uma escola etc.

### Populações infinitas

Não é possível contar seus elementos. Como exemplos temos a quantidade de porções que se pode extrair da água do mar para uma análise, temperatura medida em cada ponto de um território etc.

<font color=red>Quando os elementos de uma população puderem ser contados, porém apresentando uma quantidade muito grande, assume-se a população como infinita.</font>.

### Amostra
Subconjunto representativo da população.

Os atributos numéricos de uma população como sua média, variância e desvio padrão, são conhecidos como **parâmetros**. O principal foco da inferência estatística é justamente gerar estimativas e testar hipóteses sobre os parâmetros populacionais utilizando as informações de amostras.

## <font color=green>3.2 Quando utilizar uma amostra?</font>
***

### Populações infinitas

O estudo não chegaria nunca ao fim. Não é possível investigar todos os elementos da população.

### Testes destrutivos

Estudos onde os elementos avaliados são totalmente consumidos ou destruídos. Exemplo: testes de vida útil, testes de segurança contra colisões em automóveis.

### Resultados rápidos

Pesquisas que precisam de mais agilidade na divulgação. Exemplo: pesquisas de opinião, pesquisas que envolvam problemas de saúde pública.

### Custos elevados

Quando a população é finita mas muito numerosa, o custo de um censo pode tornar o processo inviável.

## <font color=green>3.3 Amostragem Aleatória Simples</font>
***

É uma das principais maneiras de se extrair uma amostra de uma população. A exigência fundamental deste tipo de abordagem é que cada elemeto da população tenha as mesmas chances de ser selecionado para fazer parte da amostra.

In [4]:
## A exigência fundamental deste tipo de abordagem é que cada elemeto da população tenha as mesmas
##  chances de ser selecionado para fazer parte da amostra

# selecao aleatoria pelo pandas

data.shape[0]

76840

In [5]:
data.Renda.mean()

2000.3831988547631

In [8]:
amostra = data.sample(n=100, random_state = 101)

In [9]:
amostra.shape[0]

100

In [10]:
amostra.Renda.mean()  # a amostra é representativa da população ?

2150.85

In [12]:
data.Sexo.value_counts(normalize = True)

Sexo
0    0.692998
1    0.307002
Name: proportion, dtype: float64

In [13]:
amostra.Sexo.value_counts(normalize = True)

Sexo
0    0.69
1    0.31
Name: proportion, dtype: float64

## <font color=green>3.4 Amostragem Estratificada</font>
***

É uma melhoria do processo de amostragem aleatória simples. Neste método é proposta a divisão da população em subgrupos de elementos com características similares, ou seja, grupos mais homogêneos. Com estes subgrupos separados, aplica-se a técnica de amostragem aleatória simples dentro de cada subgrupo individualmente.

## <font color=green>3.5 Amostragem por Conglomerados</font>
***

Também visa melhorar o critério de amostragem aleatória simples. Na amostragem por conglomerados são também criados subgrupos, porém não serão homogêneas como na amostragem estratificada. Na amostragem por conglomerados os subgrupos serão heterogêneos, onde, em seguida, serão aplicadas a amostragem aleatória simples ou estratificada.

Um exemplo bastante comum de aplicação deste tipo de técnica é na divisão da população em grupos territoriais, onde os elementos investigados terão características bastante variadas.

# <font color=green>4 ESTIMAÇÃO</font>
***

Estimação = Como poder estimativas dos parametros (atributos numéricos como média, desvio padrão) de uma população a partir de uma amostra !!

## <font color='red'>Problema </font>

Suponha que os pesos dos sacos de arroz de uma indústria alimentícia se distribuem aproximadamente como uma normal de **desvio padrão populacional igual a 150 g**. Selecionada uma **amostra aleatório de 20 sacos** de um lote específico, obteve-se um **peso médio de 5.050 g**. Construa um **intervalo de confiança para a média populacional** assumindo um **nível de significância de 5%**.

---

É a forma de se fazer suposições generalizadas sobre os parâmetros de uma população tendo como base as informações de uma amostra.

- **Parâmetros** são os atributos numéricos de uma população, tal como a média, desvio padrão etc.

- **Estimativa** é o valor obtido para determinado parâmetro a partir dos dados de uma amostra da população.

## <font color=green>4.1 Teorema do limite central</font>
***

> O **Teorema do Limite Central** afirma que, com o aumento do tamanho da amostra, a distribuição das médias amostrais se aproxima de uma distribuição normal com média igual à média da população e desvio padrão igual ao desvio padrão da variável original dividido pela raiz quadrada do tamanho da amostra. Este fato é assegurado para $n$ maior ou igual a 30.

# $$\sigma_\bar{x} = \frac{\sigma}{\sqrt{n}}$$

O desvio padrão das médias amostrais é conhecido como **erro padrão da média**

### Entendendo o Teorema do Limite Central

In [16]:
# assumir que quero amostrar de tamanho de 2000 do nosso dataset
n = 2000
total_de_amostras = 1500

In [17]:
# cirar dataframe vazio
amostras = pd.DataFrame()

In [18]:
for i in range(total_de_amostras):
    _ = data.Idade.sample(n)
    # resetar indice
    _.index = range(0, len(_))
    amostras['Amostra_' + str(i)] = _ 
amostras

/tmp/ipykernel_7603/2846294643.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  amostras['Amostra_' + str(i)] = _
/tmp/ipykernel_7603/2846294643.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  amostras['Amostra_' + str(i)] = _
/tmp/ipykernel_7603/2846294643.py:5: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = 

Amostra_0  Amostra_1  Amostra_2  Amostra_3  Amostra_4  Amostra_5  \
0            54         49         49         32         55         50   
1            57         38         39         46         29         49   
2            52         42         44         31         50         50   
3            27         73         62         57         60         23   
4            32         39         40         67         44         56   
...         ...        ...        ...        ...        ...        ...   
1995         35         50         26         23         41         55   
1996         29         57         47         67         49         65   
1997         50         58         37         33         52         33   
1998         41         33         47         29         43         27   
1999         31         33         46         31         50         44   

      Amostra_6  Amostra_7  Amostra_8  Amostra_9  ...  Amostra_1490  \
0            71         46         29         55  ...            47   
1            43         49         49         33  ...            47   
2            35         57         55         19  ...            60   
3            21         62         52         65  ...            49   
4            45         49         43         56  ...            38   
...         ...        ...        ...        ...  ...           ...   
1995         63         23         53         46  ...            53   
1996         38         53         35         50  ...            29   
1997         43         39         58         55  ...            46   
1998         65         29         37         60  ...            35   
1999         30         49         49         36  ...            55   

      Amostra_1491  Amostra_1492  Amostra_1493  Amostra_1494  Amostra_1495  \
0               43            21            43            60            58   
1               43            55            49            56            40   
2               49            41            23            52            38   
3               47            51            73            54            51   
4               55            39            30            49            46   
...            ...           ...           ...           ...           ...   
1995            42            36            42            51            34   
1996            42            41            42            39            51   
1997            65            55            42            26            58   
1998            79            39            37            56            41   
1999            59            51            30            50            29   

      Amostra_1496  Amostra_1497  Amostra_1498  Amostra_1499  
0               36            34            32            50  
1               52            35            36            27  
2               68            67            49            40  
3               40            58            39            53  
4               41            78            52            34  
...            ...           ...           ...           ...  
1995            48            33            36            55  
1996            37            45            41            34  
1997            49            40            31            50  
1998            63            38            35            40  
1999            42            50            56            28  

[2000 rows x 1500 columns]

In [20]:
amostras.mean()

Amostra_0       43.9530
Amostra_1       43.5165
Amostra_2       43.9965
Amostra_3       43.7930
Amostra_4       43.7935
                 ...   
Amostra_1495    44.3715
Amostra_1496    44.0725
Amostra_1497    44.1650
Amostra_1498    44.2805
Amostra_1499    44.1310
Length: 1500, dtype: float64

> O Teorema do Limite Central afirma que, **com o aumento do tamanho da amostra, a distribuição das médias amostrais se aproxima de uma distribuição normal** com média igual à média da população e desvio padrão igual ao desvio padrão da variável original dividido pela raiz quadrada do tamanho da amostra. Este fato é assegurado para n maior ou igual a 30.

In [21]:
amostras.mean().hist()

ImportError: matplotlib is required for plotting when the default backend "matplotlib" is selected.

In [22]:
data.Idade.mean()

44.07142113482561

> O Teorema do Limite Central afirma que, com o aumento do tamanho da amostra, a distribuição das médias amostrais se aproxima de uma distribuição normal **com média igual à média da população** e desvio padrão igual ao desvio padrão da variável original dividido pela raiz quadrada do tamanho da amostra. Este fato é assegurado para n maior ou igual a 30.

In [23]:
# media das medias da amostra
amostras.mean().mean()

44.071814

> O Teorema do Limite Central afirma que, com o aumento do tamanho da amostra, a distribuição das médias amostrais se aproxima de uma distribuição normal com média igual à média da população e **desvio padrão igual ao desvio padrão da variável original dividido pela raiz quadrada do tamanho da amostra**. Este fato é assegurado para n maior ou igual a 30.

# $$\sigma_\bar{x} = \frac{\sigma}{\sqrt{n}}$$

In [24]:
# desvio padrão da amostra
amostras.mean().std()

0.27178613689223474

In [28]:
# 
desvio_padrao_original = data.Idade.std()
raiz_quadrada_tamanho_amostra = np.sqrt(n)
desvio_padrao_original / raiz_quadrada_tamanho_amostra

0.27907433027404266

## <font color=green>4.2 Níveis de confiança e significância</font>
***

O **nível de confiança** ($1 - \alpha$) representa a probabilidade de acerto da estimativa. De forma complementar o **nível de significância** ($\alpha$) expressa a probabilidade de erro da estimativa.

O **nível de confiança** representa o grau de confiabilidade do resultado da estimativa estar dentro de determinado intervalo. Quando fixamos em uma pesquisa um **nível de confiança** de 95%, por exemplo, estamos assumindo que existe uma probabilidade de 95% dos resultados da pesquisa representarem bem a realidade, ou seja, estarem corretos.

O **nível de confiança** de uma estimativa pode ser obtido a partir da área sob a curva normal como ilustrado na figura abaixo.

![alt text](https://caelum-online-public.s3.amazonaws.com/1178-estatistica-parte2/01/img007.png)

## <font color=green>4.3 Erro inferencial</font>
***

O **erro inferencial** é definido pelo **desvio padrão das médias amostrais** $\sigma_\bar{x}$ e pelo **nível de confiança** determinado para o processo.

# $$e = z \frac{\sigma}{\sqrt{n}}$$

## <font color=green>4.4 Intervalos de confiança</font>
***

### Intevalo de confiança para a média da população

#### Com desvio padrão populacional conhecido

## $$\mu = \bar{x} \pm z\frac{\sigma}{\sqrt{n}}$$

#### Com desvio padrão populacional desconhecido

## $$\mu = \bar{x} \pm z\frac{s}{\sqrt{n}}$$

Formula acima:

u (acima -me) = media populacional
x barra = media da amostra
sigma = 
S = 

## <font color='blue'>Exemplo: </font>

Suponha que os pesos dos sacos de arroz de uma indústria alimentícia se distribuem aproximadamente como uma normal de **desvio padrão populacional igual a 150 g**. Selecionada uma **amostra aleatório de 20 sacos** de um lote específico, obteve-se um **peso médio de 5.050 g**. Construa um intervalo de confiança para a **média populacional** assumindo um **nível de significância de 5%**.

### Média amostral

In [29]:
n = 20
desvio_padrao_popul = 150
media_amostra = 5050
significancia = 0.05  ## logo nivel de confiança = 95%


### Nível de significância ($\alpha$)

In [30]:
significancia = 0.05  ## logo nivel de confiança = 95%

### Nível de confiança ($1 - \alpha$)

In [32]:
confianca = 1 - significancia
confianca

0.95

In [57]:
95/2


47.5

### Obtendo $z$

In [35]:
tabela_normal_padronizada[16:26]

Z       0.00    0.01    0.02    0.03    0.04    0.05    0.06    0.07    0.08  \
1.60  0.9452  0.9463  0.9474  0.9484  0.9495  0.9505  0.9515  0.9525  0.9535   
1.70  0.9554  0.9564  0.9573  0.9582  0.9591  0.9599  0.9608  0.9616  0.9625   
1.80  0.9641  0.9649  0.9656  0.9664  0.9671  0.9678  0.9686  0.9693  0.9699   
1.90  0.9713  0.9719  0.9726  0.9732  0.9738  0.9744  0.9750  0.9756  0.9761   
2.00  0.9772  0.9778  0.9783  0.9788  0.9793  0.9798  0.9803  0.9808  0.9812   
2.10  0.9821  0.9826  0.9830  0.9834  0.9838  0.9842  0.9846  0.9850  0.9854   
2.20  0.9861  0.9864  0.9868  0.9871  0.9875  0.9878  0.9881  0.9884  0.9887   
2.30  0.9893  0.9896  0.9898  0.9901  0.9904  0.9906  0.9909  0.9911  0.9913   
2.40  0.9918  0.9920  0.9922  0.9925  0.9927  0.9929  0.9931  0.9932  0.9934   
2.50  0.9938  0.9940  0.9941  0.9943  0.9945  0.9946  0.9948  0.9949  0.9951   

Z       0.09  
1.60  0.9545  
1.70  0.9633  
1.80  0.9706  
1.90  0.9767  
2.00  0.9817  
2.10  0.9857  
2.20  0.9890  
2.30  0.9916  
2.40  0.9936  
2.50  0.9952

### Obtendo $z$

![alt text](https://caelum-online-public.s3.amazonaws.com/1178-estatistica-parte2/01/img008.png)

In [36]:
## como achou o z ??? 
0.95/2  # não é 95% a area do grafico

0.475

In [37]:
# quanto falta
0.5 + (0.95/2)

0.975

In [38]:
# vai na tabela e procura o valor 0.9750 e acha o Z 
1.9 + 0.06

1.96

In [40]:
# obtendo de outra forma
z = norm.ppf(0.975)
z

1.959963984540054

### Valores de $z$ para os níveis de confiança mais utilizados

|Nível de<br>confiança|Valor da área sob<br>a curva normal| $z$ |
|:----------------:|:---------------------------------:|:---:|
|90%               |0,95                               |1,645|
|95%               |0,975                              |1,96 |
|99%               |0,995                              |2,575|

### Obtendo $\sigma_\bar{x}$

Obtendo o Sigma

In [41]:
desvio_padrao = 150
n = 20
raiz_n = np.sqrt(n)

sigma = desvio_padrao / raiz_n
sigma

33.54101966249684

### Obtendo $e$

In [42]:
# z = erro
e = z * sigma
e

65.73919054324361

65.73 gramas

### Solução 1 - Calculando o intervalo de confiança para a média

In [43]:
intervalo = (media_amostra - e,
             media_amostra + e )
intervalo

(4984.260809456757, 5115.739190543243)

Intervalo de confiança com nivel de confianca 95% - probabilidade de 95% do parametro da populacao estar neste intervalo

### Solução 2 - Calculando o intervalo de confiança para a média

In [60]:
norm.interval(confidence=0.95, loc=media_amostra, scale=sigma)

(5049.514993782489, 5050.485006217511)

mesmo resultado calculado acima !!!!

EXERCICIO

Para estimar o valor médio gasto por cada cliente de uma grande rede de fast-food, foi selecionada uma amostra de 50 clientes.

Assumindo que o valor do desvio padrão da população seja de R$ 6,00 e que esta população se distribui normalmente, obtenha a margem de erro desta estimativa para um nível de confiança de 95%.

In [56]:
desvio_p_pop = 6
n = 50
conf = 0.95
z = norm.ppf(0.975)  ## como achar o Z ?? 
sigma = desvio_p_pop / np.sqrt(n)
print(sigma)
x = z * sigma
x

0.848528137423857


1.6630845892196133

EXERCICIO

Uma amostra aleatória simples de 1976 itens de uma população normalmente distribuída, com desvio padrão populacional igual a 11, resultou em uma média amostral de 28.

Qual o intervalo de confiança de 90% para a média populacional?

In [59]:
n = 1976 
desvio = 11
media_a = 28
nivel_conf = 0.9
nivel_sig = 1 - nivel_conf
sigma = desvio / np.sqrt(n)
#
# intervalo ?

norm.interval(confidence=nivel_conf, loc=media_a, scale=sigma)

(27.592969951356466, 28.407030048643534)

# <font color=green>5 CÁLCULO DO TAMANHO DA AMOSTRA</font>
***

## <font color='red'>Problema </font>

Estamos estudando o rendimento mensal dos chefes de domicílios com renda até R$\$$ 5.000,00 no Brasil. Nosso supervisor determinou que o **erro máximo em relação a média seja de R$\$$ 10,00**. Sabemos que o **desvio padrão populacional** deste grupo de trabalhadores é de **R$\$$ 1.082,79**. Para um **nível de confiança de 95%**, qual deve ser o tamanho da amostra de nosso estudo?

## <font color=green>5.1 Variáveis quantitativas e população infinita</font>
***

# $$e = z \frac{\sigma}{\sqrt{n}}$$

#### Com desvio padrão conhecido

## $$n = \left(z\frac{\sigma}{e}\right)^2$$

#### Com desvio padrão desconhecido

## $$n = \left(z\frac{s}{e}\right)^2$$

Onde:

$z$ = variável normal padronizada

$\sigma$ = desvio padrão populacional

$s$ = desvio padrão amostral

$e$ = erro inferencial

### <font color='red'>Observações</font>

1. O desvio padrão ($\sigma$ ou $s$) e o erro ($e$) devem estar na mesma unidade de medida.

2. Quando o erro ($e$) for representado em termos percentuais, deve ser interpretado como um percentual relacionado à média.

## <font color='blue'>Exemplo: Rendimento médio</font>

Estamos estudando o rendimento mensal dos chefes de domicílios no Brasil. Nosso supervisor determinou que o **erro máximo em relação a média seja de R$\$$ 100,00**. Sabemos que o **desvio padrão populacional** deste grupo de trabalhadores é de **R$\$$ 3.323,39**. Para um **nível de confiança de 95%**, qual deve ser o tamanho da amostra de nosso estudo?

### Obtendo $\sigma$

### Obtendo $e$

### Obtendo $n$

---
---

## <font color='red'>Problema</font>

Em um lote de **10.000 latas** de refrigerante foi realizada uma amostra aleatória simples de **100 latas** e foi obtido o **desvio padrão amostral do conteúdo das latas igual a 12 ml**. O fabricante estipula um **erro máximo sobre a média populacional de apenas 5 ml**. Para garantir um **nível de confiança de 95%** qual o tamanho de amostra deve ser selecionado para este estudo?

## <font color=green>5.2 Variáveis quantitativas e população finita</font>
***

#### Com desvio padrão conhecido

## $$n = \frac{z^2 \sigma^2 N}{z^2 \sigma^2 + e^2(N-1)}$$

#### Com desvio padrão desconhecido

## $$n = \frac{z^2 s^2 N}{z^2 s^2 + e^2(N-1)}$$

Onde:

$N$ = tamanho da população

$z$ = variável normal padronizada

$\sigma$ = desvio padrão populacional

$s$ = desvio padrão amostral

$e$ = erro inferencial

## <font color='blue'>Exemplo: Indústria de refrigerantes</font>

Em um lote de **10.000 latas** de refrigerante foi realizada uma amostra aleatória simples de **100 latas** e foi obtido o **desvio padrão amostral do conteúdo das latas igual a 12 ml**. O fabricante estipula um **erro máximo sobre a média populacional de apenas 5 ml**. Para garantir um **nível de confiança de 95%** qual o tamanho de amostra deve ser selecionado para este estudo?

### Obtendo $N$

### Obtendo $z$

### Obtendo $s$

### Obtendo $e$

### Obtendo $n$

## $$n = \frac{z^2 s^2 N}{z^2 s^2 + e^2(N-1)}$$

# <font color=green>6 FIXANDO O CONTEÚDO</font>
***

## <font color='blue'>Exemplo: Rendimento médio</font>

Estamos estudando o **rendimento mensal dos chefes de domicílios com renda até R$\$$ 5.000,00 no Brasil**. Nosso supervisor determinou que o **erro máximo em relação a média seja de R$\$$ 10,00**. Sabemos que o **desvio padrão populacional** deste grupo de trabalhadores é de **R$\$$ 1.082,79** e que a **média populacional** é de **R$\$$ 1.426,54**. Para um **nível de confiança de 95%**, qual deve ser o tamanho da amostra de nosso estudo? Qual o intervalo de confiança para a média considerando o tamanho de amostra obtido?

### Construindo o dataset conforme especificado pelo problema

### Calculando o tamanho da amostra

### Calculando o intervalo de confiança para a média

### Realizando uma prova gráfica